man/man1/man1.mk

```diff
@@ -1,4 +1,8 @@
-man1_MANS += man/man1/pkcsconf.1 man/man1/pkcsicsf.1
+man1_MANS += man/man1/pkcsconf.1
+
+if ENABLE_ICSFTOK
+man1_MANS += man/man1/pkcsicsf.1
```

Comment:
When running `make distcheck` I do get the following error:
```
...
configure: creating ./config.status
config.status: error: cannot find input file: `man/man1/pkcsicsf.1.in'
make: *** [Makefile:13176: distcheck] Error 1
```

I have the feeling that this has to do with these changes in man1.mk, or the changes for `check_and_enable` in configure.ac. 

With the code on master it works.

Reply:
I compared what `make dist` produces and there are differences for the `man1` directory.
All .in files that are using `check_and_enable` in `configure.ac` are missing in the distribution tar ball:
![image](https://github.com/opencryptoki/opencryptoki/assets/28868064/1df1f8ae-7c61-4993-bd52-3fee1ffb1c8b)

I wonder if `AC_CONFIG_FILES` works if used multiple times and/or within a condition.

Reply:
> I wonder if AC_CONFIG_FILES works if used multiple times and/or within a condition.

I have found examples where `AC_CONFIG_FILES` is used multiple times, so this seems to be OK.
However, I guess `AC_CONFIG_FILES` can only be within `AM_COND_IF` or `AS_IF` to work properly, at least I see this in some examples. Can `check_and_enable` be coded using `AM_COND_IF` or `AS_IF` ?

Normally the source files of `AC_CONFIG_FILES` is automatically added to the list of files to be distributed, according to 
https://www.gnu.org/software/automake/manual/html_node/Basics-of-Distribution.html

Reply:
Just checked the final `Makefile` that automake generated, and that tracks - only the `AC_CONFIG_FILES` that are unconditional get targets generated for them that regenerates with `config.status`.

Replacing the condition handling with `AM_COND_IF` fixes this problem. I see configuration regeneration targets created for all the enabled and configured files correctly.

Reply:
Although I will need to define `DISTCHECK_CONFIGURE_FLAGS` when running `make distcheck` for AIX, otherwise it will fail to find openldap and other GNU build tools;  those are installed to `/opt/freeware` and are not added to `$PATH` by default.

Since it isn't run as part of the build process, I will document it in the readme - the [GNU docs](https://www.gnu.org/software/automake/manual/html_node/Checking-the-Distribution.html) strongly discourage from specifying flags with `AM_DISTCHECK_CONFIGURE_FLAGS`, instead making the configuration be selected dynamically at runtime.